engine/src/main/java/org/terasology/rendering/collada/ColladaLoader.java

```diff
@@ -466,14 +456,14 @@ private MD5Joint createMD5Joint(Element jointNodeElement) throws ColladaParseExc
         return md5Joint;
     }
 
-    private Quat4f[] quad4fArrayFromFloat16ArrayData(float[] inverseBindMatrixArray) {
-        Quat4f[] rotationArray = new Quat4f[inverseBindMatrixArray.length / 16];
+    private Quaternionf[] quat4fArrayFromFloat16ArrayData(float[] inverseBindMatrixArray) {
+        Quaternionf[] rotationArray = new Quaternionf[inverseBindMatrixArray.length / 16];
         for (int i = 0; i < inverseBindMatrixArray.length / 16; ++i) {
             int offset = i * 16;
-            Matrix4f matrix4f = new Matrix4f(Arrays.copyOfRange(inverseBindMatrixArray, offset, offset + 16));
-            Quat4f rotation = new Quat4f();
-            rotation.set(matrix4f);
-            rotationArray[i] = rotation;
+            Matrix4f matrix = new Matrix4f(FloatBuffer.wrap(
+                    Arrays.copyOfRange(inverseBindMatrixArray, offset, offset + 16)
+            )).transpose();
```

Comment:
I assume the old implementation worked. If so, the data in the array is stored row major and therefore the same data loaded into a joml matrix has to be transposed. What are we using the collada loader for? How can this be verified?